SVGStyleElement.sheet is undefined

rniwa opened the pull request:
#### 737ccae0f4676feb85291509f4691a2235658104
<pre>
SVGStyleElement.sheet is undefined
<a href="https://bugs.webkit.org/show_bug.cgi?id=163324">https://bugs.webkit.org/show_bug.cgi?id=163324</a>

Reviewed by Chris Dumez.

Expose sheet property on SVGStyleElement. The new behavior matches the spec, Gecko, &amp; Blink:
<a href="https://svgwg.org/svg2-draft/styling.html#InterfaceSVGStyleElement">https://svgwg.org/svg2-draft/styling.html#InterfaceSVGStyleElement</a>

* LayoutTests/imported/w3c/web-platform-tests/css/cssom/idlharness-expected.txt:
* LayoutTests/svg/dom/svg-style-element-sheet-expected.txt: Added.
* LayoutTests/svg/dom/svg-style-element-sheet.html: Added.
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/LinkStyle.idl: Added.
* Source/WebCore/html/HTMLStyleElement.idl:
* Source/WebCore/svg/SVGStyleElement.idl:

Canonical link: <a href="https://commits.webkit.org/252491@main">https://commits.webkit.org/252491@main</a>
</pre>

cdumez commented on Source/WebCore/svg/SVGStyleElement.idl:
```diff
@@ -31,4 +31,6 @@
      attribute [AtomString] DOMString media;
      [Reflect] attribute DOMString title;
+
+     readonly attribute StyleSheet sheet;
```
`StyleSheet?` per the spec:
https://www.w3.org/TR/cssom-1/#the-linkstyle-interface

rniwa replied:
oh, good point.

rniwa replied:
Also fixing `HTMLStyleElement` for completeness as we discussed on Slack.

cdumez reviewed (approved):
LGTM with minor fix.

rniwa committed:
SVGStyleElement.sheet is undefined
https://bugs.webkit.org/show_bug.cgi?id=163324

Reviewed by Chris Dumez.

Expose sheet property on SVGStyleElement. The new behavior matches the spec, Gecko, & Blink:
https://svgwg.org/svg2-draft/styling.html#InterfaceSVGStyleElement

* LayoutTests/imported/w3c/web-platform-tests/css/cssom/idlharness-expected.txt:
* LayoutTests/svg/dom/svg-style-element-sheet-expected.txt: Added.
* LayoutTests/svg/dom/svg-style-element-sheet.html: Added.
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/LinkStyle.idl: Added.
* Source/WebCore/html/HTMLStyleElement.idl:
* Source/WebCore/svg/SVGStyleElement.idl:

Canonical link: https://commits.webkit.org/252491@main
